progress/bar.go

```diff
@@ -50,44 +57,49 @@ func (b *Bar) String() string {
 
 	var pre, mid, suf strings.Builder
 
-	if b.message != "" {
+	if len(b.message) > 0 {
 		message := strings.TrimSpace(b.message)
 		if b.messageWidth > 0 && len(message) > b.messageWidth {
 			message = message[:b.messageWidth]
 		}
 
 		fmt.Fprintf(&pre, "%s", message)
-		if b.messageWidth-pre.Len() >= 0 {
-			pre.WriteString(strings.Repeat(" ", b.messageWidth-pre.Len()))
+		if padding := b.messageWidth - pre.Len(); padding > 0 {
+			pre.WriteString(strings.Repeat(" ", padding))
 		}
 
 		pre.WriteString(" ")
 	}
 
 	fmt.Fprintf(&pre, "%3.0f%% ", math.Floor(b.percent()))
-	fmt.Fprintf(&suf, "(%s/%s", format.HumanBytes(b.currentValue), format.HumanBytes(b.maxValue))
 
-	stats := b.Stats()
-	rate := int64(stats.rate)
-	if rate > 0 {
-		fmt.Fprintf(&suf, ", %s/s", format.HumanBytes(rate))
-	}
+	suf.WriteRune('(')
 
-	fmt.Fprintf(&suf, ")")
+	rate := b.rate()
+	remaining := time.Duration(int64(float64(b.maxValue-b.currentValue)/rate)) * time.Second
```

Comment:
Would this be a divide by zero on initial render?

Reply:
good catch, fixed